basecoin/src/tx/mod.rs

```diff
@@ -0,0 +1,203 @@
+}
+
+pub fn encode_auth_info(signer_info: SignerInfo, fee: Fee) -> Result<(AuthInfo, Vec<u8>), Error> {
+    let auth_info = AuthInfo {
+        signer_infos: vec![signer_info],
+        fee: Some(fee),
+        tip: None,
+    };
+
+    let mut auth_info_bytes = Vec::new();
+
+    Message::encode(&auth_info, &mut auth_info_bytes).map_err(|e| Error::Custom {
+        reason: format!("failed to encode auth info: {e}"),
+    })?;
+
+    Ok((auth_info, auth_info_bytes))
+}
+
+pub fn encode_sign_doc(
+    key_pair: KeyPair,
+    body_bytes: Vec<u8>,
+    auth_info_bytes: Vec<u8>,
+    chain_id: ChainId,
+    account_number: u64,
+) -> Result<Vec<u8>, Error> {
+    let sign_doc = SignDoc {
+        body_bytes,
+        auth_info_bytes,
+        chain_id: chain_id.to_string(),
+        account_number,
+    };
+
+    // A protobuf serialization of a SignDoc
+    let mut signdoc_buf = Vec::new();
+    Message::encode(&sign_doc, &mut signdoc_buf).unwrap();
+
+    let signed = key_pair.sign(&signdoc_buf).map_err(|e| Error::Custom {
+        reason: format!("failed to create signature: {e}"),
+    })?;
+
+    Ok(signed)
+}
+
+pub fn encode_tx_body(messages: Vec<Any>, memo: String) -> Result<(TxBody, Vec<u8>), Error> {
+    let body = TxBody {
+        messages,
+        memo,
+        timeout_height: 0_u64,
+        extension_options: Vec::<Any>::default(),
+        non_critical_extension_options: Vec::<Any>::default(),
+    };
+
+    let mut body_bytes = Vec::new();
+
+    Message::encode(&body, &mut body_bytes).map_err(|e| Error::Custom {
+        reason: format!("failed to encode transaction body: {e}"),
+    })?;
+
+    Ok((body, body_bytes))
+}
+
+pub fn encode_tx(
+    body_bytes: Vec<u8>,
+    auth_info_bytes: Vec<u8>,
+    signature_bytes: Vec<u8>,
+) -> Result<(TxRaw, Vec<u8>), Error> {
+    let tx_raw = TxRaw {
+        body_bytes,
+        auth_info_bytes,
+        signatures: vec![signature_bytes],
+    };
+
+    let mut tx_bytes = Vec::new();
+
+    Message::encode(&tx_raw, &mut tx_bytes).map_err(|e| Error::Custom {
+        reason: format!("failed to encode transaction: {e}"),
+    })?;
+
+    Ok((tx_raw, tx_bytes))
+}
+
+/// Retrieves the account sequence via gRPC client.
+pub async fn query_account(grpc_addr: Url, address: String) -> Result<BaseAccount, Error> {
+    let mut client = QueryClient::connect(grpc_addr.to_string())
+        .await
+        .map_err(|e| Error::Custom {
+            reason: format!("failed to connect to gRPC client: {e}"),
+        })?;
+
+    let request = tonic::Request::new(QueryAccountRequest { address });
+
+    let response = client.account(request).await;
+
+    let resp_account = match response
+        .map_err(|e| Error::Custom {
+            reason: format!("failed to query account: {e}"),
+        })?
+        .into_inner()
+        .account
+    {
+        Some(account) => account,
+        None => {
+            return Err(Error::Custom {
+                reason: "failed to find account".to_string(),
```

Comment:
we can use `.into()`. there are other usages.

Reply:
[d87bbd2](https://github.com/informalsystems/basecoin-rs/pull/176/commits/d87bbd21d2557ded40d99596f724469efd8678e1)